pkgs/common-updater/generic-updater.nix

```diff
@@ -15,22 +16,28 @@ let
   fileForGitCommands = "update-git-commits.txt";
 
   # shell script to update package
-  updateScript = writeScript "update-script.sh" ''
+  updateScript = writeScript "generic-update-script.sh" ''
     #! ${stdenv.shell}
     set -o errexit
     set -x
 
-    pname="$1"
-    version="$2"
-    attr_path="$3"
-    version_lister="$4"
-    ignored_versions="$5"
-    rev_prefix="$6"
-    odd_unstable="$7"
-    patchlevel_unstable="$8"
+    name="$1"
```

Comment:
At this point it might be nice to use actual named arguments like `update-source-version` does. But we can leave that for another time.